feat: Account for "VERSION" in globalConfig (#567)
UMAprotocol/UMIPs: UMIPs/umip-157.md

For example, if `l1Tokens` is "[0x123,0x456,0x789]" and `netSendAmounts` is "[1,2,3]", then the "net send amount" for token with address "0x456" is equal to "2".

## Versions

> The `ConfigStore` stores a "VERSION" value in the `globalConfig`. This is used to protect relayers and dataworkers from using outdated code when interacting with Across. The "VERSION" should be mapped to an integer string that can only increase over time. The "VERSION" is updated by calling `updateGlobalConfig` so it is emitted as an on-chain event. The block time of the event that included the "VERSION" indicates when that "VERSION" became active. Relayers should support the version at the quote timestamp for a deposit, otherwise they risk sending an invalid fill. Proposers and Disputers should support the latest version for a bundle block range to validate or propose a new bundle. This version is independent from the version included in [speed up deposit signatures](https://github.com/across-protocol/contracts-v2/blob/a8ab11fef3d15604c46bba6439291432db17e745/contracts/SpokePool.sol#L668).

Reviewer: 👌  great explanation @nicholaspai!

The following constants should reflect what is stored in the [`AcrossConfigStore`](https://etherscan.io/address/0x3b03509645713718b78951126e0a6de6f10043f5#code) contract deployed on Etherscan. This contract is owned by Across governance and acts as the source of truth for the following variables. The global variables currently stored in the above contract for this UMIP are:
- "MAX_POOL_REBALANCE_LEAF_SIZE"
- "MAX_RELAYER_REPAYMENT_LEAF_SIZE"

> - "VERSION"

Reviewer 2: Should we be explicit about whether this is a config version, or something else? I could imagine that we can tweak the config without bumping the version, but that might be counter-intuitive for anyone trying to understand the behaviour of this variable.

Author: Why would we ever want to modify the config without tweaking the version?

Reviewer: I want to make sure we're talking about the same thing:
- There's changing the values in the config (i.e. tweaking the config).
- There's changing the _structure_ of the config or how the config is _interpreted_. I would think of this as being more what the version signifies. Do y'all agree? Or are y'all talking about something else?

>   - This is a convenience to relayer and dataworker users who can compare their locally stored version against the latest version stored in the `ConfigStore` to know when to safely halt their programs and not send invalid transactions as a result of not using up to date `ConfigStore` client code.

Reviewer 2: Just throwing some ideas around - wdyt?

```suggestion
  - Across protocol version number. Supporting implementations should query this version against a locally stored value to determine compatibility with the current protocol implementation. Failure to correctly evaluate the version number may mean that filled relays are not refunded from the HubPool, and may therefore result in loss of relayer funds.
```

>   - Across protocol version number. Supporting implementations should query this value against a locally stored value to determine compatibility with the current protocol implementation. Failure to correctly evaluate the version number may mean that filled relays are not refunded from the HubPool, and may therefore result in loss of relayer funds.

Reviewer: ```suggestion
  - Across protocol version number. Supporting implementations should query this value against the value defined in their implementation to determine compatibility with the current protocol version. Failure to correctly evaluate the version number may mean that filled relays are not refunded from the HubPool, and may therefore result in loss of relayer funds.
```

The following constants should reflect what is stored in the [`AcrossConfigStore`](https://etherscan.io/address/0x3b03509645713718b78951126e0a6de6f10043f5#code) contract deployed on Etherscan. This contract is owned by Across governance and acts as the source of truth for the following variables. The global variables currently stored in the above contract for this UMIP are:
- "MAX_POOL_REBALANCE_LEAF_SIZE"
- "MAX_RELAYER_REPAYMENT_LEAF_SIZE"
- "VERSION"

>   - Across protocol version number. Supporting implementations should query this value against the value defined in their implementation to determine compatibility with the current protocol version. Failure to correctly evaluate the version number may mean that filled relays are not refunded from the HubPool, and may therefore result in loss of relayer funds. For more information go [here](#versions).

Reviewer: I think we should make sure this also applies to disputer/proposer implementations.
```suggestion
  - Across protocol version number. Supporting implementations should query this value against the value defined in their implementation to determine compatibility with the current protocol version. Failure to correctly evaluate the version number may mean that filled relays are not refunded from the HubPool, and may therefore result in loss of funds. For more information go [here](#versions).
```

   at the time of proposal.
4. No obvious griefing or manipulation of the system is being performed via this proposal.

> If the proposal is deemed invalid, return 0. If valid, return 1. Note: these values are scaled by `1e18`.

Reviewer 2: Is this just a newline change? It's not obvious in the gh diff interface.
